### PR TITLE
1069/documentation part two

### PR DESCRIPTION
**Pull request by zhuochenwang**
Second part of the documentation which focuses on the initMap() portion of the location-list.js. A lot of the methods that are called already have some description, so I just added little changes/supplements here and there. Let me know if there is an area of the code that can be worked on/documented next!

### CODE REVIEW

File: client/locations-list-map.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ const gOtherMarkers = [];
 let gPrimaryCluster = null;
 // Secondary + other markers shown in secondary cluster
 let gSecondaryCluster = null;
-
+// The longitude and latitude of the center of Google Maps
```
Comment (awong-dev) [inline client/locations-list-map.js:49]: Period at end of sentence please. :)

File: client/locations-list-map.js [hunk 2]
```diff
@@ -64,6 +64,7 @@ const gDatasetMarkers = {
   },
 };
 
+// All other datasets that are not the dataset we are currently on
```
Comment (awong-dev) [inline client/locations-list-map.js:67]: Period at end of sentence please. :)

Though this makes me think the variable is renamed. Should it be OTHER_DATASETS?

If that can make it clear enough to not need the comment...that'd even better.

File: client/locations-list-map.js [hunk 3]
```diff
@@ -231,10 +232,12 @@ function createFilters(data) {
     },
   };
 
+  // All states are included as potential filters through query and not selectable.
```
Comment (awong-dev) [inline client/locations-list-map.js:235]: "through query and not selectable." <-- this is slightly unclear to me. Are you saying

States are always included as potential filters and clients select it via the URL query parameters
even if they are not selectable?

Maybe?

File: client/locations-list-map.js [hunk 4]
```diff
@@ -231,10 +232,12 @@ function createFilters(data) {
...
   for (const state of Object.keys(data)) {
     filters.states[state] = { name: state, isSet: false };
   }
 
+  // Load all other possible whitelisted filters with their accepted, translated values
```
Comment (awong-dev) [inline client/locations-list-map.js:240]: whitelisted -> allowlisted

As an industry we should prefer allowlist/blocklist over whitelist/blacklist going forward.

File: client/locations-list-map.js [hunk 5]
```diff
@@ -1527,6 +1530,7 @@ function getDatasetFilename(dataset, countryCode) {
   return `/${dataset}-${countryCode}.json`;
 }
 
+// Grab the country codes of other countries and load their json data too
```
Comment (awong-dev) [inline client/locations-list-map.js:1533]: Period. :)

Also, it's preferable to explain the side effects of a function.  (Honestly, it's preferable to not even have global side effects, but alas...too big of a refactor).

File: client/locations-list-map.js [hunk 6]
```diff
@@ -1854,6 +1861,7 @@ function initMap(data, filters) {
 
   const mapBounds = gMap.getBounds();
 
+  // On initialization, set global variable to current center of map
```
Comment (awong-dev) [inline client/locations-list-map.js:1864]: Specify which variable.

File: client/locations-list-map.js [hunk 7]
```diff
@@ -1921,6 +1931,7 @@ function initMap(data, filters) {
 
   const legend = generateGMapDatasetLegend(onChange);
 
+  // Append the newly created legend with its event listener to top left of google Maps
```
Comment (awong-dev) [inline client/locations-list-map.js:1934]: .

File: client/locations-list-map.js [hunk 8]
```diff
@@ -1870,6 +1878,8 @@ function initMap(data, filters) {
   loadOtherCountries();
 
   // Add map control for adding additional datasets
```
Comment (awong-dev) [inline client/locations-list-map.js:1880]: Period.

File: client/locations-list-map.js (hunk 2 above)
Comment (zhuochenwang) [reply client/locations-list-map.js:67]: Yeah its always filtering out the current dataset so I wonder if I can just rename it.

Comment (awong-dev) [reply client/locations-list-map.js:67]: please do! Better variable names >> comments.

**Pull request merged**